src/AbstractOperations/conditional_operations.jl

```diff
@@ -95,29 +98,44 @@ function ConditionalOperation(c::ConditionalOperation;
     return ConditionalOperation{LX, LY, LZ}(c.operand, func, c.grid, condition, mask)
 end
 
-struct TrueCondition end
+@inline function Base.getindex(co::ConditionalOperation, i, j, k)
+    conditioned = evaluate_condition(co.condition, i, j, k, c.grid, c)
+    value = getindex(co.operand, i, j, k)
+    func_value = co.func(value)
+    return ifelse(conditioned, value, c.mask)
+end
+
+# Some special cases
+const NoFuncCO = ConditionalOperation{<:Any, <:Any, <:Any, Nothing}
+const NoConditionCO = ConditionalOperation{<:Any, <:Any, <:Any, <:Any, Nothing}
+const NoFuncNoConditionCO = ConditionalOperation{<:Any, <:Any, <:Any, Nothing, Nothing}
+
+using Base: @propagate_inbounds
+@propagate_inbounds function Base.getindex(co::NoConditionCO, i, j, k)
+    value = getindex(co.operand, i, j, k)
+    return co.func(value)
+end
+
+@propagate_inbounds Base.getindex(co::NoFuncNoConditionCO, i, j, k) = getindex(co.operand, i, j, k)
 
-@inline function Base.getindex(c::ConditionalOperation, i, j, k)
-    return ifelse(evaluate_condition(c.condition, i, j, k, c.grid, c),
-                  c.func(getindex(c.operand, i, j, k)),
-                  c.mask)
+@propagate_inbounds function Base.getindex(co::NoFuncCO, i, j, k)
+    conditioned = evaluate_condition(co.condition, i, j, k, co.grid, co)
+    value = getindex(co.operand, i, j, k)
+    return ifelse(conditioned, value, co.mask)
 end
 
-@inline evaluate_condition(condition, i, j, k, grid, args...)                = condition(i, j, k, grid, args...)
-@inline evaluate_condition(::TrueCondition, i, j, k, grid, args...)          = true
-@inline evaluate_condition(condition::AbstractArray, i, j, k, grid, args...) = @inbounds condition[i, j, k]
+# Conditions: general, nothing, array
+@inline evaluate_condition(condition, i, j, k, grid, args...) = condition(i, j, k, grid, args...)
+@inline evaluate_condition(::Nothing, i, j, k, grid, args...) = true
+@propagate_inbounds evaluate_condition(condition::AbstractArray, i, j, k, grid, args...) = condition[i, j, k]
```

Comment:
```suggestion
@inline evaluate_condition(condition::AbstractArray, i, j, k, grid, args...) = @inbounds condition[i, j, k]
```
or is it the same?

Reply:
It's not exactly the same --- with @propagate_inbounds we do still have to write `@inbouds` somewhere. But we write it at a higher level